src/search/search_engines/plugin_astar.cc

```diff
@@ -38,9 +42,14 @@ static shared_ptr<SearchEngine> _parse(OptionParser &parser) {
 
     shared_ptr<eager_search::EagerSearch> engine;
     if (!parser.dry_run()) {
-        auto temp = search_common::create_astar_open_list_factory_and_f_eval(opts);
-        opts.set("open", temp.first);
-        opts.set("f_eval", temp.second);
+        shared_ptr<Evaluator> g = make_shared<g_evaluator::GEvaluator>(
+            tasks::g_root_task);
```

Comment:
Is it necessary to use the root task here?

Reply:
We can't use the task from the search engine, because it is hard-coded (to g_root_task) in the search engine ctor, and not passed to the search engine via the options.